horizon/gopal/client.py

```diff
@@ -0,0 +1,55 @@
+import asyncio
+from typing import Optional, Awaitable, Callable
+
+from loguru import logger
+from opal_client import OpalClient
+from opal_client.engine.runner import PolicyEngineRunner
+
+from horizon.config import sidecar_config
+from horizon.gopal.runner import GopalRunner
+
+
+class GOPALClient(OpalClient):
+
+    @staticmethod
+    async def _run_engine_runner(
+        callback: Optional[Callable[[], Awaitable]],
+        engine_runner: PolicyEngineRunner,
+    ):
+        # runs the callback after policy store is up
+        engine_runner.register_process_initial_start_callbacks([callback] if callback else [])
+        async with engine_runner:
+            await engine_runner.wait_until_done()
+
+    async def _run_or_delay_for_engine_runner(
+        self, callback: Callable[[], Awaitable]
+    ):
+        if self.engine_runner:
+            # runs the callback after policy store is up
+            await self._run_engine_runner(callback, self.engine_runner)
+            return
+
+        # we do not run the policy store in the same container
+        # therefore we can immediately run the callback
+        await callback()
```

Comment:
not sure i understand this, you still need this external service which is an important dependency to become ready.

Reply:
I call it in the start_client_background_tasks

Reply:
I actually thought to do it first in the _run_or_delay_for_engine_runner function but didn't do it there eventually, so the _run_or_delay_for_engine_runner function is literally a copy paste from the parent class, will remove it